### ISSUE TITLE
[Bug]: useToastController - 'Cannot call an event handler while rendering' error when run jest tests

### ISSUE THREAD

**Issue by NataliaKravcheniaTR** [microsoft/fluentui#31701]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (8) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 4.38 GB / 31.95 GB
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.352
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://stackblitz.com/edit/acntkb?file=src%2F__tests__%2FApp.test.tsx

### Bug Description

Run tests on component which used useToastController

## Actual Behavior
'Cannot call an event handler while rendering' error

## Expected Behavior
Tests run successfully 

Note: Tests run successfully a few days ago.

### Logs

```shell
at eval (node_modules/@fluentui/react-utilities/lib-commonjs/hooks/useEventCallback.js:16:15)
      at eval (node_modules/@fluentui/react-utilities/lib-commonjs/hooks/useEventCallback.js:25:16)
      at eval (node_modules/@fluentui/react-toast/lib-commonjs/components/ToastContainer/useToastContainer.js:123:13)
      at eval (node_modules/@fluentui/react-motion/lib-commonjs/factories/createPresenceComponent.js:49:77)
      at eval (node_modules/@fluentui/react-utilities/lib-commonjs/hooks/useEventCallback.js:25:16)
      at handle.onfinish (node_modules/@fluentui/react-motion/lib-commonjs/factories/createPresenceComponent.js:91:17)
      at Object.set onfinish [as onfinish] (node_modules/@fluentui/react-motion/lib-commonjs/utils/animateAtoms.js:47:21)
      at eval (node_modules/@fluentui/react-motion/lib-commonjs/factories/createPresenceComponent.js:90:29)
      at commitHookEffectListMount (node_modules/react-dom/cjs/react-dom.development.js:23189:26)
      at commitLayoutEffectOnFiber (node_modules/react-dom/cjs/react-dom.development.js:23312:15)
      at commitLayoutMountEffects_complete (node_modules/react-dom/cjs/react-dom.development.js:24727:9)
      at commitLayoutEffects_begin (node_modules/react-dom/cjs/react-dom.development.js:24713:7)
      at commitLayoutEffects (node_modules/react-dom/cjs/react-dom.development.js:24651:3)
      at commitRootImpl (node_modules/react-dom/cjs/react-dom.development.js:26862:5)
      at commitRoot (node_modules/react-dom/cjs/react-dom.development.js:26721:5)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:26156:3)
      at flushSyncCallbacks (node_modules/react-dom/cjs/react-dom.development.js:12042:22)
      at flushActQueue (node_modules/react/cjs/react.development.js:2667:24)
      at recursivelyFlushAsyncActWork (node_modules/react/cjs/react.development.js:2633:9)
      at eval (node_modules/react/cjs/react.development.js:2545:15)
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by layershifter** [comment]
TL;DR `Toast` and other components use new motion APIs that rely on [WAAPI](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API) that is not supported by jsdom (https://github.com/jsdom/jsdom/issues/3429) (jsdom is an env Jest).

I made a previous week PR that improves compat of our internal _polyfilling_ (#31602). And similar tests are passing:

https://github.com/microsoft/fluentui/blob/825951ce337233ab8bb286a38b8995cf05d8fedb/packages/react-components/react-motion/library/src/factories/createPresenceComponent-jest.test.tsx#L47-L50

So, I was extremely confused on why this happens 🐱 This looks to be a compat issue with `@testing-library/userevent` or React, the sandbox uses version `@testing-library/user-event@14` & React 18 while we are using `@testing-library/user-event@13` & React 17.

I have built a minimal repro using React 18 (https://stackblitz.com/edit/acntkb-sru9mp) and it seems that React became _more async_ in tests.

The solution seems to use `requestAnimationFrame()` there:

https://github.com/microsoft/fluentui/blob/825951ce337233ab8bb286a38b8995cf05d8fedb/packages/react-components/react-motion/library/src/utils/animateAtoms.ts#L47-L52

It indeed fixes an issue for React 18, but breaks for React 17... 💥
